Source/Core/Core/LEDWin.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <map>
+#include <string>
+
+// from Externals/Logitech
+#include "LogitechLEDLib.h"
+
+#include "Common/Thread.h"
+#include "Common/Logging/Log.h"
+#include "Core/LED.h"
+#include "Core/HW/GCPad.h"
+#include "InputCommon/ControllerEmu.h"
+#include "InputCommon/InputConfig.h"
+
+namespace LED
+{
+
+// color table; index is used in the color mapping below!
+static const struct
+{
+	u8 red;
+	u8 green;
+	u8 blue;
+} s_colors[] =
+{
+	// 0: white (default for keys that have no other mapping)
+	{ 0xFF, 0xFF, 0xFF },
+	// 1: gray (D-Pad)
+	{ 0x95, 0x95, 0x95 },
+	// 2: green (A-Button)
+	{ 0x00, 0xFF, 0x00 },
+	// 3: red (B-Button)
+	{ 0xFF, 0x00, 0x00 },
+	// 4: yellow (C-Stick)
+	{ 0xFF, 0xFF, 0x00 },
+	// 5: purple (Z-Trigger)
+	{ 0x42, 0x00, 0xFF },
+	// 6: blue (Main Stick, since it looks nicer than white)
+	{ 0x00, 0xAA, 0xFF },
+	// 7: brown (L/R Triggers, to distinguish them from other keys)
+	{ 0x80, 0x35, 0x00 },
+};
+
+// overrides for specific named keys.
+// anything not in this list will use s_colors[0] as default color.
+// key is derived from a combination of
+// ControllerEmu::ControlGroup::name and ControllerEmu::ControlGroup::Control::name.
+// an example would be "Buttons/A" or "C-Stick/Up"
+static std::map<std::string, int> s_named_key_colors =
```

Comment:
As I mentioned, VS threw some obscure errors (like ```std::map<foo> has no matching operator[]```) when I tried that, but that might have been something else.